drivers: can: support for numaker m2l31x

cyliangtw opened the pull request:
This PR is to add support for the CAN-FD driver of Nuvoton NuMaker M2L31X series and includes the configuration of the numaker_m2l31ki board for can-api test.

Pass can-api test as:
```
------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [can_classic]: pass = 43, fail = 0, skip = 2, total = 45 duration = 0.523 seconds
SUITE PASS - 100.00% [can_stats]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.001 seconds
SUITE SKIP -   0.00% [can_transceiver]: pass = 0, fail = 0, skip = 1, total = 1 duration = 0.000 seconds
SUITE PASS - 100.00% [can_utilities]: pass = 3, fail = 0, skip = 0, total = 3 duration = 0.003 seconds
SUITE PASS - 100.00% [canfd]: pass = 12, fail = 0, skip = 0, total = 12 duration = 0.017 seconds
------ TESTSUITE SUMMARY END ------
```

cyliangtw committed:
drivers: can: support for numaker m2l31x

Add Nuvoton numaker m2l31x series can-fd controller in Kconfig.numaker
Add can-fd nodes in m2l31x.dtsi

Signed-off-by: cyliang tw <cyliang@nuvoton.com>

alexanderwachter commented on tests/drivers/can/api/boards/numaker_m2l31ki.overlay:
```diff
@@ -0,0 +1,25 @@
+};
+
+&canfd0 {
+	bus-speed = <125000>;
+	bus-speed-data = <1000000>;
```
```suggestion
```

Not needed anymore. There are defaults in Kconfig now.

cyliangtw replied:
yes, it's the same with defaults and I removed it.

henrikbrixandersen commented on tests/drivers/can/api/boards/numaker_m2l31ki.conf:
```diff
@@ -0,0 +1 @@
+CONFIG_TEST_USERSPACE=n
```
This should be handled on the board level. Otherwise we will need to have this line in many, many tests across the tree.

henrikbrixandersen replied:
Similar example of a board/MCU having too few MPU regions, handled in the board definition: https://github.com/zephyrproject-rtos/zephyr/blob/7abb9d7593b7f4d727fa1fa6440a6d87f612f99a/boards/nxp/twr_ke18f/Kconfig.defconfig#L14

cyliangtw replied:
@henrikbrixandersen ,
thanks of your advice.
Even though I tried to set `CONFIG_TEST_USERSPACE=n` in `zephyr/boards/nuvoton/numaker_m2l31ki` or in `zephyr/soc/nuvoton/numaker/m2l31x`, it's still override by `zephyr/tests/drivers/can/api/prj.conf`.
So, except `tests/drivers/can/api/boards/numaker_m2l31ki.conf`, how could  override prj.conf ?

cyliangtw replied:
@henrikbrixandersen ,
if set CONFIG_USERSPACE=n in zephyr/boards/nuvoton/numaker_m2l31ki, CONFIG_TEST_USERSPACE will not be effective.
However, loose the benefit of CONFIG_USERSPACE=y.
How about your opinion ?

cyliangtw replied:
@henrikbrixandersen ,
it's done to set the configuration on board level instead of on each test.

cyliangtw committed:
tests: drivers: can: api: support numaker_m2l31ki

Add support for Nuvoton numaker board numaker_m2l31ki.
m2l31x has 4 MPU regions and can't afford one more region
for USERSPACE, so set CONFIG_USERSPACE=n in m2l31ki_defconfig.

Signed-off-by: cyliang tw <cyliang@nuvoton.com>
